sponsorship_compassion/models/contracts.py

```diff
@@ -249,7 +249,37 @@ def create(self, vals):
                 'sponsor_id': vals.get('correspondent_id', vals['partner_id'])
             })
 
-        return super(SponsorshipContract, self).create(vals)
+        """ Generates commitment number for contracts BVRs
+        """
+        if 'commitment_number' not in vals:
+            partner_id = vals.get('partner_id')
+            correspondent_id = vals.get('correspondent_id')
```

Comment:
correspondent_id = vals.get('correspondent_id', partner_id)
So that you take the partner in case the correspondent is not defined (could happen in non-sponsorship contracts)